semgrep-core/src/engine/Match_rules.ml

```diff
@@ -239,7 +245,7 @@ let (mini_rule_of_pattern : R.t -> Pattern.t * Rule.pattern_id * string -> MR.t)
     message = "";
     severity = MR.Error;
     languages =
-      ( match r.R.languages with
+      ( match xlang with
```

Comment:
Naming nit: perhaps `xlangs` would be better? So that it is obvious this could be a list of languages

Reply:
I kind of agree given that `xlang` can carry a list of languages, although in most places it is guaranteed to carry a single language. We basically take the argument to `-lang` and pass it around as an `xlang`. For now I also prefer to be consistent with the naming used in other places of the code. We would need to ask @pad here to have the full picture.  It might be relevant to have two separate types, one to represent the languages to which a rule applies, and another to represent the actual language that was selected for a particular file.